tests/zenko_tests/node_tests/backbeat/ReplicationUtility.js

```diff
@@ -596,8 +597,9 @@ class ReplicationUtility {
                 Key: key,
                 VersionId: versionId,
             }, (err, data) => {
+                const cbOnce = jsutil.once(callback);
```

Comment:
if I understand correctly, this would happen if/because the headObject callback is invoked twice : both for the actual response and a socket timeout which happens at about the same time.

→ it seems to me a simpler fix would be just:
```
 callback => this.s3.headObject({
                Bucket: bucketName,
                Key: key,
                VersionId: versionId,
            }, jsutil.once((err, data) => {
                [...]
           })
```

However, I am wondering if there may not be a real underlying issue : like the socket timeout (on client side, i.e. test) being shorter than the timeout we have in zenko.... and maybe this could explain the (other) issues we have with these tests, what do you think?

Reply:
I am not sure it was a timeout but worth reviewing that indeed, let me check

Reply:
The current client-side timeout seem to be infinite, so I don't think we are affected by the timeout from client side:

```
const scalityS3Client = new S3({
    ...
    // disable node sdk retries and timeout to prevent InvalidPart
    // and SocketHangUp errors. If retries are allowed, sdk will send
    // another request after first request has already deleted parts,
    // causing InvalidPart. Meanwhile, if request takes too long to finish,
    // sdk will create SocketHangUp error before response.
    maxRetries: 0,
    httpOptions: { timeout: 0 },
});
```